netbox/netbox/authentication.py

```diff
@@ -386,3 +389,90 @@ def user_default_groups_handler(backend, user, response, *args, **kwargs):
             user.groups.add(*group_list)
         else:
             logger.info(f"No valid group assignments for {user} - REMOTE_AUTH_DEFAULT_GROUPS may be incorrectly set?")
+
+
+def azuread_map_groups(response, user, backend, *args, **kwargs):
+    '''
+    Map Azure AD group ID to Netbox group
+    Also set is_superuser or is_staff based on config map
+    '''
+    logger = logging.getLogger('netbox.auth.azuread_map_groups')
+
+    if not hasattr(settings, "SOCIAL_AUTH_PIPELINE_CONFIG"):
+        raise ImproperlyConfigured(
+            "Azure group mapping has been configured, but SOCIAL_AUTH_PIPELINE_CONFIG is not defined."
+        )
+
+    config = getattr(settings, "SOCIAL_AUTH_PIPELINE_CONFIG")
+    if "AZUREAD_USER_FLAGS_BY_GROUP" not in config:
+        raise ImproperlyConfigured(
+            "Azure group mapping has been configured, but AZUREAD_USER_FLAGS_BY_GROUP is not defined."
+        )
+
+    if "AZUREAD_GROUP_MAP" not in config:
+        raise ImproperlyConfigured(
+            "Azure group mapping has been configured, but AZUREAD_GROUP_MAP is not defined."
+        )
```

Comment:
Should we require both of these to be defined? I can easily see a case where the user might not care to set `AZUREAD_USER_FLAGS_BY_GROUP`. We should probably check that at least one or the other is defined, though.